### PR TITLE
Client model

### PR DESCRIPTION
**Pull request by agusiol**
Created Client , Promotion and ClientPromotion Model and added missing association. Also added some models validations.

### CODE REVIEW

File: app/controllers/clients_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class ClientsController < ApplicationController
+    before_action :set_client, only: [:show, :update, :destroy]
```
Comment (tobiaszwaszak) [inline app/controllers/clients_controller.rb:2]: similar to https://github.com/lukasz-ligizynski/Project/pull/2#discussion_r639644959

File: app/controllers/clients_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    def show
+        render json: @client
+    end
+
+    def create
+        @client = Client.new(client_params)
+        @ticket_desk = TicketDesk.find(params[:ticket_desk_id])
+        if @ticket_desk.type == "offline"
+            @client.real_user = false
+        else
```
Comment (tobiaszwaszak) [inline app/controllers/clients_controller.rb:18]: there is something wrong there in both cases you have false

File: app/controllers/reservations_controller.rb [hunk 3]
```diff
@@ -11,9 +11,9 @@ def show
 
     def create
         @screening = Screening.find(params[:screening_id])
-        #status should be somehow limited to confirmed, paied, cancelled
         @reservation = @screening.reservations.create(reservation_params)
         @reservation.ticket_desk_id = params[:ticket_desk_id]
```
Comment (tobiaszwaszak) [inline app/controllers/reservations_controller.rb:15]: you can add it to reservation_params

File: app/models/client.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Client < ApplicationRecord
+    has_many :reservations
+    has_many :clients_promotions
+    has_many :promotions, through: :clients_promotions
+
+    validates :name, :age,  presence: true
+    validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP, message: "Email invalid"  }
```
Comment (tobiaszwaszak) [inline app/models/client.rb:7]: `validates :email, presence: true, email: true` but regex is also fine

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,6 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
```
Comment (tobiaszwaszak) [inline db/seeds.rb:8]: try to avoid adding empty lines

File: app/controllers/reservations_controller.rb (hunk 3 above)
Comment (agusiol) [reply app/controllers/reservations_controller.rb:15]: Is using validate inclusion  also a good idea for this?

**Pull request closed without merging**